<launch>

  <arg name="sign" default="false"/>
  <arg name="ekf" default="false"/>
  <arg name="lane" default="false"/>
  <arg name="name" default="car1"/>
  <arg name="v" default="50"/>
  <arg name="vref" default="0.$(arg v)"/>

  <!-- speed challenge -->
  <!-- <arg name="x0" default="11.71"/>
  <arg name="y0" default="1.896"/>
  <arg name="yaw0" default="0"/>
  <arg name = "path" default="speedrun"/> -->

  <!-- speedcurve -->
  <!-- <arg name="x0" default="3.66"/>
  <arg name="y0" default="12.186"/>
  <arg name="yaw0" default="3.14159"/>
  <arg name = "path" default="speedcurve"/> -->

  <!-- highway_east -->
  <!-- <arg name="x0" default="2.48"/>
  <arg name="y0" default="11.788"/>
  <arg name="yaw0" default="0"/>
  <arg name = "path" default="highway_east"/> -->

  <!-- bus_lane -->
  <!-- <arg name="x0" default="15.3"/>
  <arg name="y0" default="6.196"/>
  <arg name="yaw0" default="-1.5708"/>
  <arg name = "path" default="bus_lane"/> -->

  <!-- parking -->
  <!-- <arg name="x0" default="0.35"/>
  <arg name="y0" default="2.726"/>
  <arg name="yaw0" default="-1.5708"/>
  <arg name = "path" default="parking"/> -->

  <!-- oneway -->
  <arg name="x0" default="4.76"/>
  <arg name="y0" default="2.706"/>
  <arg name="yaw0" default="-1.5708"/>
  <arg name = "path" default="oneway"/>

  <param name="x0" value="$(arg x0)"/>
  <param name="y0" value="$(arg y0)"/>
  <param name="yaw0" value="$(arg yaw0)"/>

  <arg name="real" default="false"/>
  <arg name="subModel" default="false"/>
  <arg name = "rate" default="300"/>
  <param name="pathName" value="$(arg path)"/>
  <param name="vrefInt" value="$(arg v)"/>
  <arg name = "gmapping" default="false"/>
  <param name="gmapping" value="$(arg gmapping)"/>
  <arg name = "amcl" default="false"/>
  <param name="amcl" value="$(arg amcl)"/>
  <arg name = "lidarOdom" default="false"/>
  <param name="lidarOdom" value="$(arg lidarOdom)"/>
  <arg name = "dashboard" default="true"/>
  <param name="dashboard" value="$(arg dashboard)"/>
  <arg name = "realsense_imu" default="false"/>
  <param name="realsense_imu" value="$(arg realsense_imu)"/>
  <arg name = "kb" default="false"/>
  <param name="kb" value="$(arg kb)"/>
  <arg name = "hw" default="1.33"/>
  <param name="hw" value="$(arg hw)"/>
  <arg name = "gps" default="false"/>
  <param name="gps" value="$(arg gps)"/>
  <arg name = "pub_wpts" default="true"/>
  <param name="pub_wpts" value="$(arg pub_wpts)"/>
  <arg name = "debug_level" default="5"/>
  <param name="debug_level" value="$(arg debug_level)"/>
  <rosparam command="load" file="$(find control)/scripts/config/mpc_config$(arg v).yaml" />
  <rosparam command="load" file="$(find control)/config/tunable_params.yaml" />
  <rosparam command="load" file="$(find control)/scripts/config/mpc_config_park.yaml" />

  <!-- <node pkg="control" type="path2.py" name="path_planner" output="screen"></node>  -->
  <node pkg="control" type="controller" name="$(anon mpc_controller)" output="screen"> 
    <param name="sign" value="$(arg sign)"/>
    <param name="ekf" value="$(arg ekf)"/>
    <param name="lane" value="$(arg lane)"/>
    <param name="name" value="$(arg name)"/>
    <param name="/robot_name" value="$(arg name)"/>
    <param name="vref" value="$(arg vref)"/>
    <param name="x0" value="$(arg x0)"/>
    <param name="y0" value="$(arg y0)"/>
    <param name="yaw0" value="$(arg yaw0)"/>
    <param name="real" value="$(arg real)"/>
    <param name="/real" value="$(arg real)"/>
    <param name="subModel" value="$(arg subModel)"/>
    <param name="rate" value="$(arg rate)"/>
  </node>
</launch>
